collections/course_records.ts

```diff
@@ -34,7 +33,24 @@ course_records.allow({
 /* Schema */
 declare var SimpleSchema: any;
 
+
 if(Meteor.isServer) {
+	Meteor.publish('course-records', function() {
+		const user = Meteor.users.findOne(this.userId);
```

Comment:
Check if userid is valid BEFORE you query. Otherwise you can pass in a null object which returns all of the userids, effectively making every guest a random authorized user!!